checks/permissions.go

```diff
@@ -244,66 +261,73 @@ func calculateScore(result permissionCbData) int {
 	// Start with a perfect score.
 	score := float32(checker.MaxResultScore)
 
-	// If no top level permissions are defined, all the permissions
-	// are enabled by default, hence "all". In this case,
-	if permissionIsPresentInTopLevel(result, "all") {
-		if permissionIsPresentInRunLevel(result, "all") {
-			// ... give lowest score if no run level permissions are defined either.
-			return checker.MinResultScore
+	// Retrieve the overall results.
+	for _, perms := range result.workflows {
+		// If no top level permissions are defined, all the permissions
+		// are enabled by default, hence "all". In this case,
+		if permissionIsPresentInTopLevel(perms, "all") {
+			if permissionIsPresentInRunLevel(perms, "all") {
+				// ... give lowest score if no run level permissions are defined either.
+				return checker.MinResultScore
+			}
+			// ... reduce score if run level permissions are defined.
+			score -= 0.5
 		}
-		// ... reduce score if run level permissions are defined.
-		score--
-	}
 
-	// status: https://docs.github.com/en/rest/reference/repos#statuses.
-	// May allow an attacker to change the result of pre-submit and get a PR merged.
-	// Low risk: -0.5.
-	if permissionIsPresent(result, "statuses") {
-		score -= 0.5
-	}
+		// status: https://docs.github.com/en/rest/reference/repos#statuses.
+		// May allow an attacker to change the result of pre-submit and get a PR merged.
+		// Low risk: -0.5.
+		if permissionIsPresent(perms, "statuses") {
```

Comment:
Nit: make these strings - "all", "statuses", "checks" etc. constants or better yet define a new type for these strings. It helps avoid bugs due to typos or definition mismatches.

Reply:
done, thanks.